test/executor.test.js

```diff
@@ -352,41 +386,51 @@ describe('executor', function() {
       delete process.env.npm_package_config_port;
     });
 
-    function bootWithDefaults() {
+    function bootWithDefaults(done) {
       app = loopback();
       boot.execute(app, someInstructions({
-        config: {
+        application: {
           port: undefined,
           host: undefined,
         },
-      }));
+      }), done);
     }
 
-    it('should apply env passed in option object', function() {
-      boot.execute(app, someInstructions({ env: 'custom_env' }));
-      expect(app.get('env')).to.equal('custom_env');
+    it('should apply env passed in option object', function(done) {
+      boot.execute(app, someInstructions({ env: 'custom_env' }), function(err) {
+        if (err) return done(err);
+        expect(app.get('env')).to.equal('custom_env');
+        done();
+      });
     });
 
-    it('should honor host and port', function() {
-      function assertHonored(portKey, hostKey) {
+    it('should honor host and port', function(done) {
+      function assertHonored(portKey, hostKey, done) {
         process.env[hostKey] = randomPort();
         process.env[portKey] = randomHost();
-        bootWithDefaults();
-        assert.equal(app.get('port'), process.env[portKey], portKey);
-        assert.equal(app.get('host'), process.env[hostKey], hostKey);
-        delete process.env[portKey];
-        delete process.env[hostKey];
+        bootWithDefaults(function(err) {
+          if (err) return done(err);
+          assert.equal(app.get('port'), process.env[portKey], portKey);
+          assert.equal(app.get('host'), process.env[hostKey], hostKey);
+          delete process.env[portKey];
+          delete process.env[hostKey];
+          done();
+        });
       }
 
-      assertHonored('OPENSHIFT_SLS_PORT', 'OPENSHIFT_NODEJS_IP');
-      assertHonored('npm_config_port', 'npm_config_host');
-      assertHonored('npm_package_config_port', 'npm_package_config_host');
-      assertHonored('OPENSHIFT_SLS_PORT', 'OPENSHIFT_SLS_IP');
-      assertHonored('VCAP_APP_PORT', 'VCAP_APP_HOST');
-      assertHonored('PORT', 'HOST');
+      async.eachSeries([
+        { port: 'OPENSHIFT_SLS_PORT', host: 'OPENSHIFT_NODEJS_IP' },
+        { port: 'npm_config_port', host: 'npm_config_host' },
+        { port: 'npm_package_config_port', host: 'npm_package_config_host' },
+        { port: 'OPENSHIFT_SLS_PORT', host: 'OPENSHIFT_SLS_IP' },
+        { port: 'VCAP_APP_PORT', host: 'VCAP_APP_HOST' },
+        { port: 'PORT', host: 'HOST' },
+      ], function(config, done) {
```

Comment:
We usually use `next` as the callback argument name. It makes it clear that the next item will be processed when the callback is invoked. It also prevents confusion about which `done` callback is invoked, whether the inner `done` for `async.eachSeries` or the outer `done` for the unit test.